sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -613,6 +625,57 @@ Increment and decrement operator of annotated_ref. Increment/Decrement the objec
 referenced by this wrapper via ``T``'s Increment/Decrement operator.
 
 The annotations are applied when the object `T` is loaded and stored to the memory.
+
+a|
+[source,c++]
+----
+template <class A, class B>
+auto friend operatorOP(const A& a, const B& b) ->
+		   decltype(remove_ann_ref<A> OP remove_ann_ref<B>);
+----
+a|
+Where [code]#OP# is: [code]#pass:[+]#, [code]#-#,[code]#*#, [code]#/#, [code]#%#, [code]#+<<+#, [code]#>>#, [code]#&#, [code]#\|#, [code]#^#.
+
+Defines a hidden friend operator `OP` overload for types `A` and `B`.
+
+When `A` is an `annotated_ref<T, PropertyListT>`, `remove_ann_ref<A>` returns `T`, otherwise
+returns `A`. Same applied to `B`. This operator is invoked only when at least one of the `A`
+and `B` is an `annotated_ref<T, PropertyListT>`.
+
+Let `operatorOP` denotes the operator used and `X`, `Y` denote the type returned by `remove_ann_ref<A>`
+and `remove_ann_ref<B>`.
+
+The overloaded operator `operatorOP` utilizes `operatorOP(X, Y)` and
+is available only if `operatorOP(X, Y)` is well formed. The value and result
+is the same as the result of `operatorOP(X, Y)` applied to the objects of
+type `X` and `Y`.
+
+If type `A` or `B` is an `annotated_ref<T, PropertyListT>`, the annotations from `PropertyListT`
+are applied when the object `a` or `b` is loaded from memory. 
+
+a|
+[source,c++]
+----
+template <class A> auto friend operatorOP(const A& a) ->
```

Comment:
If this was specified as:
```
auto operatorOP() -> decltype(OP std::declval<T>());
```
The description would be easier because you don't need remove_ann_ref.